test/unit/update_versions_file_test.rb

```diff
@@ -44,7 +44,7 @@ class UpdateVersionsFileTest < ActiveSupport::TestCase
       end
 
       should "include platform release" do
-        expected_output = "rubyrubyruby 0.0.1-jruby,0.0.1 qw212r\n"
+        expected_output = "rubyrubyruby 0.0.1,0.0.1-jruby qw212r\n"
```

Comment:
There's also this "ordering change" that I'm not sure where it comes from :man_shrugging:

Reply:
Maybe related to https://github.com/rubygems/compact_index/commit/63187755018e11ffc22b6e56d8463d13898b1a0e?

Reply:
yes, simi is correct. The order doesn't matter at the moment, but allow me to explain this to ensure all of us are on the same page.
You may already know that bundler uses the checksum of the local gem info file and checksum reported by the versions file to decide if it should make a request to rg.org to update it's gem info file.
The info checksum reported in the versions endpoint for the gem will change when a gem releases or yanks a version. info checksum as reported by versions endpoint always has to be one calculated for the latest update (created_at/id column of versions table), so that the client can see checksum mismatch and fetches the latest changes.
Before [this change](https://github.com/rubygems/rubygems.org/pull/1454/files), we were sorting versions by [number, platform], and it meant we were using the checksum of the last version by number and platform (not last created at) making the client miss info updates if someone release `0.1.0` after `2.0.1`. 

Above linked PR would the second instance of us trying to fix something broken in `compact_index` by overriding the behavior in rubygems.org. Most plausible reason for doing this seems to be avoiding the hassle of releasing a new version. 
It makes me question if compact_index really has to be separate repo.

Reply:
Thanks for explaining, makes sense.

Note that `bundler` also uses the `compact_index` gem for testing, I guess that's why it is in a separate repo.

Reply:
So, as per the code comment in the commit you linked to, we could now get rid of https://github.com/rubygems/rubygems.org/pull/1454/files#diff-2cf34b93c3257e00201d866038d19583R91-R93, right?

Reply:
Code has moved, actually, it's here now:

https://github.com/rubygems/rubygems.org/blob/995069b09bc138fe662ff78af9b5a368b7c64371/app/models/gem_info.rb#L64-L65

Reply:
> bundler also uses the compact_index gem for testing

I would consider re-evaluating if [these need to use compact_index API](https://github.com/rubygems/bundler/blob/bf7df421ee6311fcc17e6dc46d204236f33b4b3f/spec/support/artifice/compact_index.rb#L94-L116) or we can just hardcode the response.

> we could now get rid of https://github.com/rubygems/rubygems.org/pull/1454/files#diff-2cf34b93c3257e00201d866038d19583R91-R93, right?

We could. I would lean towards not changing anything tho. Given the checksum mismatch we are seeing, we still seem to be figuring out all the edge cases.

Reply:
Sure. I only found out recently that rubygems.org also used this gem, I thought it was just a little gem to mock rubygems.org responses. When I found about it, it felt nice that we were using the same thing but if you feel there's value in replacing it, I'm fine with re-evaluating it, sure.

Reply:
rubygems.org is the only user if you ask me. I see `vcr_cassettes` in bundler repo, they could be used for mocking rubygems.org response. are you sure you are using the latest compact_index version?

Reply:
> there's value in replacing it

value being not keeping a dependency you don't need, even if it is just a test/development dependency.

Reply:
> are you sure you are using the latest compact_index version?

No, we are using 0.11.0, we haven't upgraded yet.

> value being not keeping a dependency you don't need, even if it is just a test/development dependency.

It depends on the alternative, removing a dependency is not good per se. But as I said, I'm opening to exploring alternatives :+1:

Reply:
I just tried upgrading and our specs crash with the new version: https://github.com/rubygems/rubygems/pull/3615. Needs investigation.

Reply:
> `NoMethodError: undefined method 'gsub' for #<Gem::Requirement:0x00005628ab6bfb28>`

from the following:
```
 def ruby_version_line
      join_multiple(ruby_version)
 end
```
AFAIU, compact_index only supports string as requirement. Removing compact_index from bundler is going to be harder than I assumed :sweat_smile: 
will look into removing above mentioned patch from rubygems.org. We should probably move more things to compact_index (like generating new versions file) both from rubygems.org and bundler if we do have to keep compact_index the gem around.

Reply:
That last issue was solved in bundler, and my original comment was explained too, so marking this as resolved :)